processor/src/errors.rs

```diff
@@ -28,15 +28,27 @@ pub enum ExecutionError {
     DivideByZero(u32),
     EventError(String),
     Ext2InttError(Ext2InttError),
-    FailedAssertion(u32, Felt),
+    FailedAssertion {
+        clk: u32,
+        err_code: u32,
+        err_msg: Option<&'static str>,
```

Comment:
Should we use `Option<String>` here? I wonder if static strings may be too limiting in some contexts.